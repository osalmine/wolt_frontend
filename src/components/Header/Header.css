@media (min-width: 992px) {
	.navbar-center {
		position: absolute;
		left: 50%;
		transform: translatex(-50%);
	}
}

.navbar-brand-text {
	font-size: 28px !important;
}

.navbar-custom-text {
	font-size: 22px;
}