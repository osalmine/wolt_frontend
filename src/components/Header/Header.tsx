import { Section } from '../RestaurantSection/RestaurantSection';
import { Navbar, Nav } from 'react-bootstrap';
import './Header.css';
import 'bootstrap/dist/css/bootstrap.min.css';

interface Sections {
	sections: Section[]
}

interface Props {
	sectionsProp: Sections;
}

const Header: React.FC<Props> = ({ sectionsProp }) => {
	return <>
		<Navbar collapseOnSelect expand="lg" bg="light" variant="light" fixed="top" className="py-3">
			<Navbar.Brand href="/" className="navbar-brand-text">Discovery</Navbar.Brand>
			<Navbar.Toggle aria-controls="responsive-navbar-nav" />
			<Navbar.Collapse id="responsive-navbar-nav">
				<Nav className="mr-auto navbar-center navbar-custom-text">
					{sectionsProp.sections.map((section, i) => 
						<Nav.Link key={i} href={'#' + section.title.replace(/ /g,'')}>{section.title}</Nav.Link>)}
				</Nav>
			</Navbar.Collapse>
		</Navbar>
	</>;
};

export default Header;