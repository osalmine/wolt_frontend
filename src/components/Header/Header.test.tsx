import Header from './Header';
import inputJson from '../../discovery_page.json';
import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('Header component tests', () => {
	test('renders header from inputJson', () => {
		render(<Header sectionsProp={inputJson} />);
	
		const linkElement = screen.getByText(/Popular Restaurants/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	test('renders with no input', () => {
		const input = {
			"sections": []
		};
		render(<Header sectionsProp={input} />);
	
		const linkElement = screen.getByText(/Discovery/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	test('matches snapshot', () => {
		const tree = renderer.create(<Header sectionsProp={inputJson} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});