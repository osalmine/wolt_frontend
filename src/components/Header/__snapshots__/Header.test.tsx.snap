// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests matches snapshot 1`] = `
<nav
  className="py-3 navbar navbar-expand-lg navbar-light bg-light fixed-top"
>
  <a
    className="navbar-brand-text navbar-brand"
    href="/"
  >
    Discovery
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto navbar-center navbar-custom-text navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        className="nav-link"
        data-rb-event-key="#PopularRestaurants"
        href="#PopularRestaurants"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Popular Restaurants
      </a>
      <a
        className="nav-link"
        data-rb-event-key="#NewRestaurants"
        href="#NewRestaurants"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        New Restaurants
      </a>
      <a
        className="nav-link"
        data-rb-event-key="#NearbyRestaurants"
        href="#NearbyRestaurants"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Nearby Restaurants
      </a>
    </div>
  </div>
</nav>
`;
